sdk/eventhub/event-hubs/src/impl/awaitableQueue.ts

```diff
@@ -25,13 +27,23 @@ export class AwaitableQueue<T> {
   /**
    * Returns a Promise that will resolve with the next item in the queue.
    */
-  public shift(): Promise<T> {
+  public shift(options?: AbortOptions): Promise<T> {
     const item = this._items.shift();
     if (typeof item !== "undefined") {
       return Promise.resolve(item);
     }
 
-    return new Promise<T>((resolve) => this._resolvers.push(resolve));
+    return createAbortablePromise<T>(
+      (resolve) => {
+        return this._resolvers.push(resolve);
+      },
```

Comment:
nitpick:
```suggestion
      (resolve) => this._resolvers.push(resolve),
```

or even just:
```suggestion
      this._resolvers.push,
```

Reply:
picking option 1 since it is more clear